test/attr/attr_availability.swift

```diff
@@ -1179,3 +1179,47 @@ func testMultipleTrailingClosures(_ x: TypeWithTrailingClosures) {
   // expected-note@-1 {{use 'TypeWithTrailingClosures.variadicTrailingClosures(a:b:c:)' instead}} {{3-27=x.variadicTrailingClosures}} {{28-29=}} {{none}}
   x.variadicTrailingClosures() {} _: {} _: {}
 }
+
+struct UnavailableSubscripts { 
+  @available(*, unavailable, renamed: "subscript(new:)")
+  subscript(old index: Int) -> Int { 3 } // expected-note * {{'subscript(old:)' has been explicitly marked unavailable here}}
+  @available(*, unavailable, renamed: "subscript(new:)")
+  func getValue(old: Int) -> Int { 3 } // expected-note * {{'getValue(old:)' has been explicitly marked unavailable here}}
+
+  subscript(new index: Int) -> Int { 3 }
+
+  @available(*, unavailable, renamed: "getAValue(new:)")
+  subscript(getAValue index: Int) -> Int { 3 } // expected-note * {{'subscript(getAValue:)' has been explicitly marked unavailable here}}
+  func getAValue(new: Int) -> Int { 3 }
+
+  @available(*, unavailable, renamed: "subscript(arg1:arg2:arg3:)")
+  subscript(_ argg1: Int, _ argg2: Int, _ argg3: Int) -> Int { 3 } // expected-note * {{'subscript(_:_:_:)' has been explicitly marked unavailable here}}
+
+  @available(*, deprecated, renamed: "subscript(arg1:arg2:arg3:)")
+  subscript(argg1 argg1: Int, argg2 argg2: Int, argg3 argg3: Int) -> Int { 3 }
+
+  @available(*, deprecated, renamed: "subscript(arg1:arg2:arg3:)")
+  subscript(only1 only1: Int, only2 only2: Int) -> Int { 3 }
+
+  subscript(arg1 arg1: Int, arg2 arg2: Int, arg3 arg3: Int) -> Int { 3 }
+
+  func testUnavailableSubscripts(_ x: UnavailableSubscripts) { 
+    _ = self[old: 3] // expected-error {{'subscript(old:)' has been renamed to 'subscript(new:)'}} {{14-17=new}}
+    _ = x[old: 3] // expected-error {{'subscript(old:)' has been renamed to 'subscript(new:)'}} {{11-14=new}}
```

Comment:
Could you please add a couple of examples where subscript is renamed into a function?

Reply:
Ah, I completely missed `[` fix-it for that case...